### ISSUE TITLE
500 error when X-Client-State header contains nonsense

### ISSUE THREAD

**Issue by michielbdejong** [mozilla-services/syncto#4]
When you send an x-client-state header of `'[object Object]'`, the syncto server responds with a 500 Internal Server Error, instead of with a more meaningful response code.

Trace from syncto when this happens:

``` bash
.venv/bin/cliquet --ini config/development.ini migrate
INFO:venusian:syncto 1.0.0.dev0 starting. 
.venv/bin/pserve config/development.ini --reload
Starting subprocess with file monitor
Starting server in PID 54212.
serving on http://0.0.0.0:8000
ERROR:venusian:"GET   /v1/buckets/syncto/collections/tabs/records" ? (? ms) 400 Client Error: Bad Request lang=en-US,en;q=0.5; exception=Traceback (most recent call last):
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/pyramid/tweens.py", line 21, in excview_tween
    response = handler(request)
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/pyramid/router.py", line 163, in handle_request
    response = view_callable(context, request)
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/pyramid/config/views.py", line 596, in __call__
    return view(context, request)
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/pyramid/config/views.py", line 329, in attr_view
    return view(context, request)
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/pyramid/config/views.py", line 305, in predicate_wrapper
    return view(context, request)
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/pyramid/config/views.py", line 355, in rendered_view
    result = view(context, request)
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/pyramid/config/views.py", line 501, in _requestonly_view
    response = view(request)
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/cornice/service.py", line 569, in wrapper
    response = view_(request)
  File "/Users/Michiel/syncto/syncto/views/collection.py", line 49, in collection_get
    sync_client = SyncClient(bid_assertion, client_state)
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/sync/client.py", line 35, in __init__
    self._authenticate(bid_assertion, client_state, tokenserver_url)
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/sync/client.py", line 54, in _authenticate
    raw_resp.raise_for_status()
  File "/Users/Michiel/syncto/.venv/lib/python2.7/site-packages/requests/models.py", line 851, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
HTTPError: 400 Client Error: Bad Request; uid=None; errno=None; agent=Mozilla/5.0 (Mobile; rv:42.0) Gecko/20100101 Firefox/42.0; authn_type=None
```

**Comment by michielbdejong** [comment]
Same happens when the Sync servers responds with `401 Client Error: Unauthorized`, for instance (I think) if you use an expired assertion.

**Comment by michielbdejong** [comment]
Especially for 401 errors, this is likely to block some work I have planned for this sprint. Anything I can do to help?

**Comment by Natim** [comment]
This is my next priority this week.

**Comment by michielbdejong** [comment]
Awesome! :) Let me know if I can help in any way.

**Comment by Natim** [comment]
Fixed in #12, @michielbdejong feel free to review.

**Comment by michielbdejong** [comment]
Great! Will play with it today.
